sys/ztimer64/ztimer64.c

```diff
@@ -129,27 +129,21 @@ static int _del_entry_from_list(ztimer64_clock_t *clock, ztimer64_base_t *entry)
 {
     DEBUG("_del_entry_from_list()\n");
     assert(_is_set((ztimer64_t *)entry));
-    if (clock->first == entry) {
-        /* special case: removing first entry */
-        clock->first = entry->next;
-        entry->next = 0;
-        return 1;
-    }
-    else {
-        ztimer64_base_t *pos = clock->first;
+    int res = 1;
+    ztimer64_base_t **pos = &clock->first;
 
-        while (pos->next) {
-            if (pos->next == entry) {
-                pos->next = entry->next;
-                break;
-            }
-            pos = pos->next;
+    while (*pos) {
+        if (*pos == entry) {
+            *pos = entry->next;
```

Comment:
when a element is removed from a singly linked list a next pointer has to be modified, this does not.